igb_client/client.py

```diff
@@ -0,0 +1,189 @@
+            self._base_url.format(view="jobboards"),
+        )
+        if not resp.ok:
+            raise IGBClientError("Error when connecting to IGB")
+
+        job_boards = resp.json()["HAPI"]["jobboards"]
+        return [
+            JobBoard(
+                name=job_board["jobboard"].get("name"),
+                klass=job_board["jobboard"].get("class"),
+                logo=job_board["jobboard"].get("logo"),
+                compliance=job_board["jobboard"].get("compliance"),
+                classifications=job_board["jobboard"].get("classifications"),
+            )
+            for job_board in job_boards
+        ]
+
+    def detail(self, job_board: str) -> Optional[JobBoard]:
+        resp = self.session.get(
+            self._base_url.format(view=f"jobboards/{job_board}")
+        )
+        if not resp.ok:
+            raise IGBClientError("Error when connecting to IGB")
+
+        job_board = resp.json()["HAPI"]["jobboard"]
+        if job_board:
+            job_board = parse_igb_xml_payload(job_board)
+            return JobBoard(
+                name=job_board.get("name"),
+                klass=job_board.get("class"),
+                moc=job_board.get("MOC"),
+                instructions=job_board.get("instructions"),
+                ofccp=job_board.get("OFCCP"),
+                classifications=job_board.get("OFCCP", {}).get("classifications"),
+                facets=job_board.get("facets"),
+            )
+
+
+class IGBFacets(IGBClientBase):
+    def get_board_facets(
+            self,
+            credentials: Optional[ContractCredential],
+            facet_name: str,
+            term: str = None,
+    ) -> List[Dict]:
+        autocomplete_url = self._base_url.format(
+            view=f"/{credentials.job_board.klass}/facet/{facet_name}/custom"
+        )
+
+        params = {}
+
+        if credentials:
+            params = self.encrypt_credentials(credentials).credentials
+
+        if term:
+            params["term"] = term
+
+        resp = self.session.post(
+            autocomplete_url,
+            json={"params": params},
+        )
+
+        if not resp.ok:
+            raise IGBClientError("Error when connecting to IGB")
+
+        return [
+            {"key": item["key"], "label": item["label"]}
+            for item in parse_igb_xml_payload(resp.json()).get("options", [])
+        ]
+
+    def validate(
+            self, credentials: ContractCredential, facet_name: str, keys: list
+    ) -> bool:
+        validate_url = self._base_url.format(
+            view=f"jobboards/{credentials.job_board.klass}/facet/{facet_name}/custom/validate"
+        )
+
+        params = self.encrypt_credentials(credentials).credentials
+
+        params["keys"] = [{"key": key} for key in keys]
+
+        resp = self.session.post(
+            validate_url,
+            json={"params": params},
+        )
+
+        if not resp.ok:
+            raise IGBClientError("Error when connecting to IGB")
```

Comment:
I think it would be nice to print the http response code in the IGBClientError exception.
Exemple: If the response code is 400, the connection is OK but the payload is wrong